File: ts/128/1.ts
/**
 * 枚举法
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const twoSum = function(nums, target) {
  for (let i = 0; i < nums.length; i++) {
    let another = target - nums[i]
    for (let j = i + 1; j < nums.length; j++) {
      if(nums[j] === another) {
        return [i, j]
      }
    }
  }
}


/**
 * 哈希表
 */
function twoSum1(nums: number[], target: number): number[] {
  const map = new Map<number, number>()
  for (let i = 0; i < nums.length; i++) {
    const another:number = target - nums[i]
    if (map.has(another)) {
      return [map.get(another), i]
    }
    map.set(nums[i], i)
  }
  return []
}

File: c/128/1.c
#include <stdlib.h>

// 枚举法
int* twoSum(int* nums, int numsSize, int target, int* returnSize){
  for (int i = 0; i < numsSize; i++) {
    int another = target - nums[i];
    for (int j = i + 1; j < numsSize; j++) {
      if (nums[j] == another) {
        int* res = (int*)malloc(sizeof(int) * 2);
        res[0] = i;
        res[1] = j;
        *returnSize = 2;
        return res;
      }
    }
  };
  *returnSize = 0;
  return 0;
}